packages/protocol/contracts/signal/SignalService.sol

```diff
@@ -39,13 +38,13 @@ contract SignalService is AuthorizableContract, ISignalService {
     struct Hop {
         address signalRootRelay;
         bytes32 signalRoot;
-        bytes storageProof;
+        bytes[] storageProof;
```

Comment:
these are breaking changes as relayer proof data needs to be re-formatted.

Reply:
Actually i like it better than translating the encoded proof to a format like [adopt_optimism_new_trie_code](https://github.com/taikoxyz/taiko-mono/tree/adopt_optimism_new_trie_code) bc. it will create some on-chain overhead.

Reply:
I like this approach too, but if we merge this PR then the bridge relayer will have to be changed and we have a lot more work to do. Let the bridge relayer fix the existing bugs first.